scripts/main.component.ts

```diff
@@ -20,18 +21,36 @@ import { REPLComponent, REPLDelegate } from "./repl.component"
   templateUrl: "../html/main.component.html",
   providers: [MenuService, PluginService]
 })
-export class MainComponent implements OnInit, MenuDelegate, REPLDelegate {
+export class MainComponent implements OnInit, MenuDelegate, REPLDelegate, EditorDelegate {
   constructor(private menu: MenuService, private pluginService: PluginService) {}
+  
+  selectedElements : Set<DrawableGraph | DrawableNode | DrawableEdge> =
+    new Set<DrawableGraph | DrawableNode | DrawableEdge>();
+  selectElement(element : DrawableGraph | DrawableNode | DrawableEdge) {
+    this.selectedElements.add(element);
+  }
+  deselectElement(element : DrawableGraph | DrawableNode | DrawableEdge) {
+    this.selectedElements.delete(element);
+  }
+  clearSelected() {
+    this.selectedElements.clear();
+  }
+
+
   ngOnInit(): void {
     this.menu.setDelegate(this);
     this.repl.setDelegate(this);
+    this.graphEditor.delegate = this;
```

Comment:
I vote either the way I did it, or `addDelegate`/`removeDelegate`. `set` implies theres only one